File: csharp/grains/Grains.cs
using System;

public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64) {
            throw new ArgumentOutOfRangeException("Square must be between 1 and 64.");
        }
        return (ulong)Math.Pow(2, n - 1);
    }

    public static ulong Total()
    {
        ulong res = 0;
        for (int i = 0; i < 64; i++) {
            res += (ulong)Math.Pow(2, i);
        }
        return res;
    }
}

File: java/grains/src/main/java/Grains.java
import java.math.BigInteger;

class Grains {

    BigInteger one = new BigInteger("1");

    BigInteger grainsOnSquare(final int square) {
        if (square < 1 || square > 64) {
            throw new IllegalArgumentException("square must be between 1 and 64");
        }
        return one.shiftLeft(square - 1);
    }

    BigInteger grainsOnBoard() {
        return one.shiftLeft(64).subtract(one);
    }

}
